source/common/util/generate-filename.ts

```diff
@@ -14,19 +14,25 @@
 
 import replaceStringVariables from './replace-string-variables'
 import generateId from './generate-id'
-import { v4 as uuid } from 'uuid'
 
 /**
  * Generates a new filename based on the configured filename pattern.
+ * @param {string} filenamePattern - The pattern for generating filenames.
+ * @param {string} idGenPattern - The pattern for generating unique IDs.
+ * @param {string} ext - The file extension to be used (default: 'md').
  * @return {string} The new filename.
  */
-export default function generateFilename (filenamePattern: string, idGenPattern: string): string {
+export default function generateFilename (filenamePattern: string, idGenPattern: string, ext: string ): string {
```

Comment:
Suggestion to strongly type the extension:

```suggestion
export default function generateFilename (filenamePattern: string, idGenPattern: string, ext: 'md'|'json'|'yaml'|'tex' ): string {
```